src/main/java/net/snowflake/client/jdbc/SnowflakeConnectionV1.java

```diff
@@ -567,11 +567,21 @@ public SQLXML createSQLXML() throws SQLException {
 
   @Override
   public boolean isValid(int timeout) throws SQLException {
-    // TODO: run query here or ping
     if (timeout < 0) {
       throw new SQLException("timeout is less than 0");
+    } else if (isClosed) {
+      return false;
+    } else {
+      try {
+        Statement statement = this.createStatement();
+        statement.setQueryTimeout(timeout);
+        statement.execute("select 1");
+        statement.close();
+      } catch (SQLException ex) {
+        return false;
+      }
+      return true;
```

Comment:
`A value of 0 indicates a timeout is not applied to the database operation.` Just to double check is this our behavior when the timeout is 0?

Reply:
Yes, isValid() directly passes the timeout value to setQueryTimeout(), and when queryTmeout has a value of 0, no timeout is applied to the query. https://docs.snowflake.com/en/user-guide/jdbc-parameters.html#querytimeout